plugins/glustershd/rest.go

```diff
@@ -121,3 +125,92 @@ func selfhealInfoHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, &info.Bricks)
 
 }
+
+func selfHealHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var healType string
+	heal, ok := r.URL.Query()["type"]
+	if ok {
+		switch heal[0] {
+		case "index":
+			healType = "1"
+		case "full":
+			healType = "2"
+		default:
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "heal type can only be either index or full")
+			return
+		}
+	} else {
+		healType = "1"
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		if err == transaction.ErrLockTimeout {
+			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+	defer txn.Done()
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		if err == gderrors.ErrVolNotFound {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, gderrors.ErrVolNotFound)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrVolNotStarted)
+		return
+	}
+
+	// Check if self heal is already enabled
+	if !isHealEnabled(volinfo) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "self heal option is disabled for this volume")
+		return
+	}
+
+	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
+		logger.WithError(err).WithField("key", "volinfo").Error("failed to set key in transaction context")
```

Comment:
I think we don't need WithFiled in logging here. you  can simplify this 
```
logger.WithError(err).Error("failed to set volinfo in transaction context")
```

Reply:
WithField is better. That's the convention we follow. It makes logs parse-able by future tools.

When it comes to txn context in particular, it's logged internally by the framework. This will remove a lot of repetitive logging like this one.

Reply:
Actually, we are not logging with WithFiled in case of a setting transaction fails in gd2.

Reply:
@vpandey-RH You can remove the log here as it is done internally again